### PR TITLE
[chore] Update `FeatureModel` to skip duplicated table when generating `table_id_feature_job_settings`

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR updates feature model's root validator to generate `table_id_feature_job_settings` without duplicated table IDs.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/query_graph/graph.py [hunk 1]
```diff
@@ -263,7 +265,13 @@ def extract_table_id_feature_job_settings(
         node_table_id_iterator = self.iterate_group_by_node_and_table_id_pairs(
             target_node=target_node
         )
+        table_ids = set()
         for group_by_node, table_id in node_table_id_iterator:
+            if keep_first_only and table_id in table_ids:
+                # skip duplicate table
```
Comment (YS-L) [inline featurebyte/query_graph/graph.py:271]: In which cases will we want to keep duplicate tables?

**Review by YS-L — APPROVED** [review]

File: featurebyte/query_graph/graph.py (hunk 1 above)
Comment (gee-senbong) [reply featurebyte/query_graph/graph.py:271]: When we validate whether the duplicate tables have the same setting or not in https://github.com/featurebyte/featurebyte/blob/main/featurebyte/service/feature.py#L117-L148.

Comment (gee-senbong) [reply featurebyte/query_graph/graph.py:271]: 🤔 Maybe we should make the `keep_first_only` default to `True` & update the parameter there.

Comment (YS-L) [reply featurebyte/query_graph/graph.py:271]: I think either default value is fine. Would be helpful to have a comment on this.

**Pull request merged**